src/expression.d

```diff
@@ -13976,7 +13985,10 @@ public:
 
     override Expression toBoolean(Scope* sc)
     {
-        e2 = e2.toBoolean(sc);
+        auto ex2 = e2.toBoolean(sc);
```

Comment:
Currently, if a semantic analysis is done in error gagging section, the analyzed original AST might be reused. Therefore, if a sub AST generates `ErrorExp`, it should not be stored back into the original AST. If not, the stored `ErrorExp` would appear in the whole semantic analysis result, and of course the result AST will become invalid.

Until we completely change the gagging mechanism, we should use such the idiomatic code.